src/main/java/me/oyurimatheus/nossomercadolivre/products/SendQuestionToSellersEmailListener.java

```diff
@@ -11,27 +11,33 @@
 
     private final EmailService sendEmail;
     private final EmailRepository emailRepository;
+    private final QuestionRepository questionRepository;
 
     SendQuestionToSellersEmailListener(EmailService sendEmail,
-                                       EmailRepository emailRepository) {
+                                       EmailRepository emailRepository,
+                                       QuestionRepository questionRepository) {
         this.sendEmail = sendEmail;
         this.emailRepository = emailRepository;
+        this.questionRepository = questionRepository;
     }
 
     @EventListener
-    void listen(QuestionEvent question) {
+    void listen(QuestionEvent questionEvent) {
 
         // TODO: Apply I18N in this messages?
         var subject = " You have a new question";
-        var body = question.getTitle() + " in " + question.getProductUri();
+        var body = questionEvent.getTitle() + " in " + questionEvent.getProductUri();
 
-        Email email = Email.to(question.getSellersEmail())
-                           .from(question.getPossibleBuyer())
+        Email email = Email.to(questionEvent.getSellersEmail())
+                           .from(questionEvent.getPossibleBuyer())
                            .subject(subject)
                            .body(body)
                            .build();
 
         sendEmail.send(email);
         emailRepository.save(email);
+        Question question  = questionRepository.findById(questionEvent.getId()).get();
```

Comment:
você preferiu fazer esse código para não associar o email com a pergunta? O problema desse design aqui é que ele não é aderente a funcionalidade :). Lá não tinha dito que uma pergunta tem um email. Só pedia para gravar o email relacionado a aquela pergunta.

Reply:
Mas quando eu faço um `emailRepository.save(email)` eu já não to salvando o email que foi enviado? Eu tinha entendido que precisava relacionar o email de alguma forma a nível de banco a pergunta...

Reply:
A ideia é que o email seja relacionado com a pergunta e não o contrário :). 

Mas o problema aqui do código é que você, de maneira camuflada, altera uma referência alheia. Você recebeu um id de uma pergunta, carregou e alterou aqui. Mas se você tem o id da pergunta nesse evento que roda em memória, é pq alguma pergunta já foi carregada antes :).

Reply:
Se quiser manter o relacionamento dentro da pergunta, mande bala, mas aí você teria que realizar a alteração de outro lugar :). Mas acho que a causa raiz é realmente o fato de você ter modelado de um jeito que jogou contra você.

Reply:
Sim... queria deixar o email genérico, sem precisar linkar com o produto ou outra entidade... Talvez desse pra deixar genérica a ponto de ter dois campos um com o nome da entidade e outro com o id dela, mas acho que assim só tornaria mais complexo.